pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java

```diff
@@ -183,6 +184,22 @@ private void preGenesisSubscription(final BigInteger lastProcessedBlockNumber) {
             });
   }
 
+  private SafeFuture<Void> waitForEth1ProviderToBeInSync() {
+    return eth1Provider
+        .ethSyncing()
+        .thenCompose(
+            isSyncing -> {
+              if (isSyncing) {
+                LOG.info(
+                    "Eth1DepositManager failed to get the head of Eth1: still syncing. Retrying in {} seconds.",
+                    Constants.ETH1_SYNCING_RETRY_TIMEOUT.toSeconds());
+                return asyncRunner
+                    .getDelayedFuture(Constants.ETH1_SYNCING_RETRY_TIMEOUT)
+                    .thenCompose(__ -> waitForEth1ProviderToBeInSync());
+              } else return SafeFuture.COMPLETE;
```

Comment:
We can simplify this a bit with `asyncRunner.runAfterDelay`.  And we always use braces on code blocks.
```suggestion
                return asyncRunner.runAfterDelay(
                    this::waitForEth1ProviderToBeInSync, Constants.ETH1_SYNCING_RETRY_TIMEOUT);
              } else {
                return SafeFuture.COMPLETE;
              }
```